geode-wan/src/main/java/org/apache/geode/internal/cache/wan/serial/SerialGatewaySenderImpl.java

```diff
@@ -117,15 +117,8 @@ private void start(boolean cleanQueues) {
   }
 
   protected AbstractGatewaySenderEventProcessor createEventProcessor(boolean cleanQueues) {
-    AbstractGatewaySenderEventProcessor eventProcessor;
-    if (getDispatcherThreads() > 1) {
-      eventProcessor = new RemoteConcurrentSerialGatewaySenderEventProcessor(
-          SerialGatewaySenderImpl.this, getThreadMonitorObj(), cleanQueues);
-    } else {
-      eventProcessor = new RemoteSerialGatewaySenderEventProcessor(SerialGatewaySenderImpl.this,
-          getId(), getThreadMonitorObj(), cleanQueues);
-    }
-    return eventProcessor;
+    return new RemoteConcurrentSerialGatewaySenderEventProcessor(
+        SerialGatewaySenderImpl.this, getThreadMonitorObj(), cleanQueues);
```

Comment:
Would it be possible to address the underlying problem with using `RemoteSerialGatewaySenderEventProcessor` rather than simply not using it? There appears to be an NPE thrown when calling `GatewaySenderMXBean.isConnected()` on the `GatewaySenderMXBean` proxy in `ListGatewayCommand` when `dispatcher-threads=1`, which this change does not directly address. It would be better to address the root cause of this issue if possible rather than just avoiding it, I think.

Reply:
Hi @DonalEvans ,
I agree with you, I will close this PR and open a new one with new fix.